examples/simple/src/comments/PostReferenceInput.js

```diff
@@ -0,0 +1,134 @@
+    };
+
+    handleCloseCreate = () => {
+        this.setState({ showCreateDialog: false });
+    };
+
+    handleCloseShow = () => {
+        this.setState({ showPreviewDialog: false });
+    };
+
+    handleSave = post => {
+        this.setState({ showCreateDialog: false, new_post_id: post.id });
+    };
+
+    render() {
+        const { showCreateDialog, showPreviewDialog, new_post_id } = this.state;
+        const { classes, translate, post_id, ...props } = this.props;
+
+        return (
+            <Fragment>
+                <ReferenceInput {...props} defaultValue={new_post_id}>
+                    <SelectInput optionText="title" />
+                </ReferenceInput>
+                <Button
+                    className={classes.button}
+                    onClick={this.handleNewClick}
+                >
+                    New
+                </Button>
+                {post_id && (
+                    <Button
+                        className={classes.button}
+                        onClick={this.handleShowClick}
+                    >
+                        Show
+                    </Button>
+                )}
+                <Dialog
+                    fullWidth
+                    open={showCreateDialog}
+                    onClose={this.handleCloseCreate}
+                    aria-label={translate('simple.create-post')}
+                >
+                    <DialogTitle>{translate('simple.create-post')}</DialogTitle>
+                    <DialogContent>
+                        <PostQuickCreate
+                            onCancel={this.handleCloseCreate}
+                            onSave={this.handleSave}
+                            basePath="/posts"
+                            formName="post-create"
+                            resource="posts"
+                        />
+                    </DialogContent>
+                </Dialog>
+                <Dialog
+                    fullWidth
+                    open={showPreviewDialog}
+                    onClose={this.handleCloseShow}
+                    aria-label={translate('simple.create-post')}
+                >
+                    <DialogTitle>{translate('simple.create-post')}</DialogTitle>
+                    <DialogContent>
+                        <PostPreview
+                            id={post_id}
+                            basePath="/posts"
+                            resource="posts"
+                        />
+                    </DialogContent>
+                    <DialogActions>
+                        <Button onClick={this.handleCloseShow}>
+                            {translate('simple.action.close')}
+                        </Button>
+                    </DialogActions>
+                </Dialog>
+            </Fragment>
+        );
+    }
+}
+
+const formSelector = formValueSelector('record-form');
+
+const mapStateToProps = state => ({
+    post_id: formSelector(state, 'post_id'),
```

Comment:
why do you need it? react-admin injects the `record` to inputs, too.

Reply:
`record` !== form values

Reply:
then why don't you use redux-form's `<Field>` instead?

Reply:
I don't get this

Reply:
ah, got it! Much simpler indeed :+1: